src/SimulationRunner.cc

```diff
@@ -565,10 +567,22 @@ void SimulationRunner::ProcessSystemQueue()
         this->postUpdateStartBarrier->Wait();
         if (this->postUpdateThreadsRunning)
         {
-          system->PostUpdate(this->currentInfo, this->entityCompMgr);
+          auto terminate = this->threadsToTerminate.find(parentEntity);
+          if (terminate != this->threadsToTerminate.end()) {
+            terminate->second--;
+            if (terminate->second == 0) {
+              this->threadsToTerminate.erase(terminate);
```

Comment:
:warning: Threads to terminate is not thread safe. Rework this bit.

Reply:
[0089730](https://github.com/gazebosim/gz-sim/pull/2232/commits/008973093d0e87d592d1876f3fcf5d6b363d11d1)